### PR TITLE
feat(init): Catch GitHub 403 errors and provide a more useful error message

### PR DESCRIPTION
**Pull request by AKPWebDesign**
### Description

When attempting to `firebase init` a Hosting project, then choosing to set up a GitHub workflow, it is possible to provide a repo which is in an organization that the user has access to, but that they have not provided access to the Firebase CLI OAuth app for. If this happens, all the user gets is a fairly unhelpful error message: image
  
Instead, I'd propose that we catch these 403 errors and provide a more useful message to the user, like this: 
image

### Scenarios Tested

I manually tested this change, checking that the new messaging shows when trying to connect to a GitHub repository that is in an organization which has not been granted access for.

### CODE REVIEW

File: src/init/features/hosting/github.ts
Comment (samtstern) [inline src/init/features/hosting/github.ts:427]: Minor comment: could you combine these into a single error message?  It's ok if it's long, we have plenty of multi-line error messages.

**Comment by samtstern** [comment]
@AKPWebDesign thank you for the contribution!  I think this is a great change, but I will leave this to @jhuleatt or @davideast  or @jamesdaniels  to approve since they own this Action flow.

File: src/init/features/hosting/github.ts
Comment (AKPWebDesign) [reply src/init/features/hosting/github.ts]: Alright, I updated this to be one error message instead of two, and updated the screenshot in the PR to match.

**Comment by davideast** [comment]
Just seeing this now! I'll get to it in a day or two!

**Comment by AKPWebDesign** [comment]
Just wanted to bump this real quick, in case it's been forgotten about!

**Comment by samtstern** [comment]
@davideast can you give this another look?

**Comment by AKPWebDesign** [comment]
It's been a little while. Anything I can do to help this along?

**Review by samtstern — APPROVED** [review]

**Comment by samtstern** [comment]
@AKPWebDesign sorry about this! I guess @davideast missed this. LGTM, if you can just merge `master` into your branch to resolve the conflicts I am happy to merge this PR>

**Comment by AKPWebDesign** [comment]
@samtstern no worries! I just merged `master` in and everything looks good assuming the builds pass!

**Comment by samtstern** [comment]
@AKPWebDesign merged, thanks for your contribution and your patience!

**Pull request merged**
